### PR TITLE
Story/q20 5

### PR DESCRIPTION
**Pull request by RenukaVarma19**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x ] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [ x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x ] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x ] I am not reformatting any files in the wrong format or without cause. 
- [ x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x ] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x ] I have added test cases where appropriate
- [ x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

[Q20-5] made additional changes according to the review comments and fixed the test case.
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: quadriga/src/main/java/edu/asu/diging/quadriga/service/impl/CollectionManager.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class CollectionManager implements ICollectionManager {
+
+    @Autowired
+    private CollectionRepository collectionRepo;
+
+    
+    /**
+     * Creates a new Collection instance and stores it in mongodb
+     * 
+     * @param collectionForm   Form data that needs to be added to database
```
Comment (jdamerow) [inline quadriga/src/main/java/edu/asu/diging/quadriga/service/impl/CollectionManager.java]: this is incorrect

File: quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        
+    }
+
+    @Test
+    public void test_addCollection_success() {
+    	Collection collection1 = new Collection();
+        collection1.setName("name");
```
Comment (jdamerow) [inline quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerTest.java]: indentation

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
- "I have added Javadoc/documentation where appropriate" should be checked.
- please reopen existing PRs next time

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
Let's also add a page that shows all collections. I will create new tickets for deleting/editing collections, but without an overview page that shows all collections it's hard to tell if something went wrong.

**Comment by RenukaVarma19** [comment]
Added the requested changes on indentation

**Comment by jdamerow** [comment]
what about:
> Let's also add a page that shows all collections. I will create new tickets for deleting/editing collections, but without an overview page that shows all collections it's hard to tell if something went wrong.

**Comment by RenukaVarma19** [comment]
Added Show Collections page to show all the collections

File: quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    private CollectionRepository collectionRepo;
+
+    @RequestMapping(value = "/auth/collections/add", method = RequestMethod.GET)
+    public String get(Model model) {
+        model.addAttribute("collectionForm", new CollectionForm());
+        return "admin/user/addCollection";
+    }
+    
+	
+    @RequestMapping(value="/auth/collections/showcollection",method=RequestMethod.GET)
```
Comment (jdamerow) [inline quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java]: this should just be `/auth/collections`

File: quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            return "admin/user/addCollection";
+        }
+        Collection collection=new Collection();
+        collection.setName(collectionForm.getName());
+        collection.setDescription(collection.getDescription());
+        collectionManager.addCollection(collection);
+        
+        redirectAttrs.addFlashAttribute("alert_type", "success");
+        redirectAttrs.addFlashAttribute("alert_msg", "Collection has been added.");
+        return "redirect:/auth/collections/add";
```
Comment (jdamerow) [inline quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java]: should redirect to the collections list page

File: quadriga/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 5]
```diff
@@ -50,6 +50,14 @@
 	          		<a th:href="@{/admin/user/list}" >Users</a>
 	          	</li>
 	          	
+	          	<li sec:authorize="hasRole('USER')" role="presentation">
+	          		<a th:href="@{/auth/collections/add}" >Add Collection</a>
+	          	</li>
+	          	
+	          	<li sec:authorize="hasRole('USER')" role="presentation">
+	          		<a th:href="@{/auth/collections/showcollection}" >Show Collections</a>
+	          	</li>
```
Comment (jdamerow) [inline quadriga/src/main/webapp/WEB-INF/views/layouts/main.html]: this should be a drop down menu "Collections" with two menu items (See all and Add collection)

File: quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    public String get(Model model) {
+        model.addAttribute("collectionForm", new CollectionForm());
+        return "admin/user/addCollection";
+    }
+    
+	
+    @RequestMapping(value="/auth/collections/showcollection",method=RequestMethod.GET) 
+    public String showCollection(Model model) {
+    	model.addAttribute("collections",collectionRepo.findAll()); 
+    	return "showcollection";
```
Comment (jdamerow) [inline quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java]: the template should be in the same folder as the addcollection template

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by RenukaVarma19** [comment]
Added the requested changes

File: quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    private CollectionRepository collectionRepo;
+
+    @RequestMapping(value = "/auth/collections/add", method = RequestMethod.GET)
+    public String get(Model model) {
+        model.addAttribute("collectionForm", new CollectionForm());
+        return "admin/user/addCollection";
+    }
+    
+	
+    @RequestMapping(value="/auth/collections",method=RequestMethod.GET)
```
Comment (jdamerow) [inline quadriga/src/main/java/edu/asu/diging/quadriga/web/AddCollectionController.java]: should be in its own controller

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: quadriga/src/main/java/edu/asu/diging/quadriga/web/ShowCollectionController.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+@Controller
+public class ShowCollectionController {
+
+	@Autowired
+	private CollectionRepository collectionRepo;
+
+	@RequestMapping(value="/auth/collections",method=RequestMethod.GET) 
+	public String showCollection(Model model) {
+		model.addAttribute("collections",collectionRepo.findAll());
```
Comment (jdamerow) [inline quadriga/src/main/java/edu/asu/diging/quadriga/web/ShowCollectionController.java]: let's paginate this (if there are 100 collections you don't want them all on one page).

File: quadriga/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 9]
```diff
@@ -50,6 +50,21 @@
 	          		<a th:href="@{/admin/user/list}" >Users</a>
 	          	</li>
 	          	
+	          	<li sec:authorize="hasRole('USER')" role="presentation">
```
Comment (jdamerow) [inline quadriga/src/main/webapp/WEB-INF/views/layouts/main.html]: the drop down menu button shouldn't look like it's selected. It should have a white background like the others and only get a grey background on hover.

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**